tests/core/test_bsdf.py

```diff
@@ -0,0 +1,353 @@
+        for btdf1data, btdf2data in zip(bsdf1.btdf, bsdf2.btdf):
+            test_list.append(compare_bsdf_data_point(btdf1data, btdf2data))
+    if all(test_list):
+        return True
+    else:
+        return False
+
+
+def test_anisotropic_bsdf(speos: Speos):
+    """Unit test for anisotropic bsdf class."""
+    initial_bsdf = create_anisotropicbsdf(speos)
+    bsdf_path = Path(test_path) / "Test_Lambertian_bsdf.anisotropicbsdf"
+    initial_bsdf.save(bsdf_path)
+
+    # check save
+    assert does_file_exist(str(bsdf_path))
+
+    # compare loaded with created
+    exported_bsdf = AnisotropicBSDF(speos, bsdf_path)
+    assert compare_anisotropic_bsdf(initial_bsdf, exported_bsdf)
+
+    # remove transmission
+    exported_bsdf.has_transmission = False
+    assert not compare_anisotropic_bsdf(initial_bsdf, exported_bsdf)
+
+    # compare only reflective
+    initial_bsdf.has_transmission = False
+    assert compare_anisotropic_bsdf(initial_bsdf, exported_bsdf)
+
+    # change spectrum incidence
+    exported_bsdf.spectrum_incidence = np.radians(5)
+    exported_bsdf.spectrum_anisotropy = np.radians(2)
+    assert not compare_anisotropic_bsdf(initial_bsdf, exported_bsdf)
+
+    # test reset
+    exported_bsdf.reset()
+    initial_bsdf.reset()
+    assert compare_anisotropic_bsdf(initial_bsdf, exported_bsdf)
+
+    # test commit True/false
+    # change value
+    exported_bsdf.spectrum_incidence = [np.radians(5), np.radians(5)]
+    exported_bsdf.spectrum_anisotropy = [np.radians(2), np.radians(2)]
+
+    # save non changed file
+    bsdf_path2 = Path(test_path) / "Test_Lambertian_bsdf2"
+    bsdf_path2 = exported_bsdf.save(bsdf_path2, commit=False)
+    assert does_file_exist(str(bsdf_path2))
+
+    # save changed file
+    bsdf_path3 = Path(test_path) / "Test_Lambertian_bsdf3"
+    bsdf_path3 = exported_bsdf.save(bsdf_path3, commit=True)
+    assert does_file_exist(str(bsdf_path3))
+
+    # load and compare files
+    bsdf2 = AnisotropicBSDF(speos, bsdf_path2)
+    bsdf3 = AnisotropicBSDF(speos, bsdf_path3)
+    assert compare_anisotropic_bsdf(initial_bsdf, bsdf2)
+    assert not compare_anisotropic_bsdf(bsdf2, bsdf3)
+    remove_file(str(bsdf_path))
+    remove_file(str(bsdf_path2))
+    remove_file(str(bsdf_path3))
+
+
+def test_anisotropic_bsdf_interpolation_enhancement(speos: Speos):
+    """Unit test for anisotropic bsdf interpolation class."""
+    # test automatic interpolation enhancement
+    input_file = Path(test_path) / "Gaussian Fresnel 10 deg.anisotropicbsdf"
+    output_file = Path(test_path) / "Gaussian Fresnel 10 deg interpolation test.anisotropicbsdf"
+    initial_bsdf = AnisotropicBSDF(speos=speos, file_path=input_file)
+
+    # test indices setting and cones data generation
+    automatic_interpolation_settings = initial_bsdf.interpolation_enhancement(
+        index_1=1.0, index_2=1.5
+    )
+    assert automatic_interpolation_settings.index1 == 1
+    assert automatic_interpolation_settings.index2 == 1.5
+
```

Comment:
setting refractiv indexes after wards missing
```suggestion
automatic_interpolation_settings.index1 = 1,5
automatic_interpolation_settings.index2 = 1

assert automatic_interpolation_settings.index1 == 1,5
assert automatic_interpolation_settings.index2 == 1
```